src/daemon/daemon.cpp

```diff
@@ -2023,36 +2198,63 @@ void mp::Daemon::create_vm(const CreateRequest* request, grpc::ServerWriter<Crea
 
                 reply.set_create_message("Configuring " + name);
                 server->Write(reply);
-                auto vendor_data_cloud_init_config =
-                    make_cloud_init_vendor_config(*config->ssh_key_provider, request->time_zone(), config->ssh_username,
-                                                  config->factory->get_backend_version_string().toStdString());
-                auto meta_data_cloud_init_config = make_cloud_init_meta_config(name);
-                auto user_data_cloud_init_config = YAML::Load(request->cloud_init_user_data());
-                prepare_user_data(user_data_cloud_init_config, vendor_data_cloud_init_config);
 
-                std::string mac_addr;
-                while (true)
+                // To generate the network interfaces, we need to check first for repetition the MAC addresses
+                // which were specified by the user, and add them to the set of used MAC addresses. But if it
+                // happens that we have repeated addresses, we need to remove addresses we inserted on this step.
+                // Only after doing this we will be able to generate the rest of the addresses.
```

Comment:
I don't think adding and removing is the right thing. Instead, we should either:

- only add after confirming it is not there yet, or
- work with a temporary set and replace once we know there were no repetitions

Reply:
The reason why I implemented that way is that it permitted me to use `generate_unused_mac_address`. The fix was to add a new parameter to this function, which is another set. Then the function generates a new address which is not used in the daemon and does not belong to the set. So, done.